src/test/lexer/incremental.ts

```diff
@@ -229,6 +229,95 @@ describe(`Lexer.Incremental`, () => {
             expect(state.lines[1].text).to.equal("bXr");
             expect(state.lines[2].text).to.equal("baz");
         });
+
+        it(`lineTerminator maintained on single line change`, () => {
+            let range: Lexer.Range = {
+                start: {
+                    lineNumber: 1,
+                    lineCodeUnit: 1,
+                },
+                end: {
+                    lineNumber: 1,
+                    lineCodeUnit: 2,
+                },
+            };
+
+            let original: string = `foo\nbar\nbaz`;
+            const state: Lexer.State = expectLexerUpdateRangeOk(
+                original,
+                "X",
+                range
+            );
+
+            let modified = original.replace("bar", "bXr");
```

Comment:
either inline into `expect(...).equals(...)` or move it closer to the expect stmt.